### PR TITLE
Refactory: move api and client-go to standalone repos

### PR DESCRIPTION
**Pull request by zhaohuabing**
Move Aeraki API to  https://github.com/aeraki-mesh/api
Move Aeraki API client go lib to https://github.com/aeraki-mesh/client-go
The API and client-go packages inside Aeraki repo will be keep for a while to avoid any breaking changes to projects that depend on Aeraki.

All the Istio dependencies have also been updated to 1.16.5 .

### CODE REVIEW

File: internal/plugin/dubbo/authz/builder/builder.go [hunk 1]
```diff
@@ -22,17 +22,17 @@ import (
 
 	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 
-	dubborulepb "github.com/aeraki-mesh/aeraki/api/dubbo/v1alpha1"
-	dubboapi "github.com/aeraki-mesh/aeraki/client-go/pkg/apis/dubbo/v1alpha1"
-	dubboclient "github.com/aeraki-mesh/aeraki/client-go/pkg/clientset/versioned/typed/dubbo/v1alpha1"
+	dubborulepb "github.com/aeraki-mesh/api/dubbo/v1alpha1"
```
Comment (tanjunchen) [inline internal/plugin/dubbo/authz/builder/builder.go:25]: 这些包的导入顺序建议按照
golang 内置包、第三方依赖、项目依赖包顺序，其他的也同理。

File: go.mod [hunk 2]
```diff
@@ -2,126 +2,149 @@ module github.com/aeraki-mesh/aeraki
...
-	k8s.io/client-go v0.22.2
-	sigs.k8s.io/controller-runtime v0.10.2
+	go.uber.org/atomic v1.10.0
+	golang.org/x/net v0.12.0
+	golang.org/x/sync v0.3.0
+	google.golang.org/grpc v1.54.0
+	google.golang.org/protobuf v1.30.0
+	istio.io/api v1.19.0-alpha.1
+	istio.io/client-go v1.18.1
+	istio.io/gogo-genproto v0.0.0-20230705213215-8f59b5a7ddde
```
Comment (tanjunchen) [inline go.mod:62]: 我们还使用 istio.io/gogo-genproto 这个包？我看 istio 现在不依赖这个包了。

File: go.mod [hunk 3]
```diff
@@ -2,126 +2,149 @@ module github.com/aeraki-mesh/aeraki
...
-	k8s.io/api v0.22.2
-	k8s.io/apimachinery v0.22.2
-	k8s.io/client-go v0.22.2
-	sigs.k8s.io/controller-runtime v0.10.2
+	go.uber.org/atomic v1.10.0
+	golang.org/x/net v0.12.0
+	golang.org/x/sync v0.3.0
+	google.golang.org/grpc v1.54.0
+	google.golang.org/protobuf v1.30.0
+	istio.io/api v1.19.0-alpha.1
```
Comment (tanjunchen) [inline go.mod:60]: istio.io/api  建议使用 istio 1.16.x 版本，目前 istio 最新的 tag 版本是 1.16.7 .

Comment (zhaohuabing) [reply go.mod:60]: Thanks for the review. I'll address these comments in another PR.

Comment (zhaohuabing) [reply go.mod:60]: This import is generated by `go tidy`, and has been replaced by 1.16.7

```
replace istio.io/api => istio.io/api v0.0.0-20230518153929-d0aebaa77ab8
```

File: go.mod (hunk 2 above)
Comment (zhaohuabing) [reply go.mod:62]: It's used in `api`, which is deprecated.

**Pull request merged**